build.rs

```diff
@@ -99,6 +101,32 @@ fn fetch_libipt() {
     env::set_current_dir(&Path::new("..")).unwrap();
 }
 
+// Checks if the CPU supports Intel Processor Trace.
+// We use this to decide whether to run the perf_pt backend tests. Although this would be better as
+// a runtime check, it's OK since we won't distribute the test binary.
+fn cpu_supports_pt() -> bool {
+    const LEAF: u32 = 0x07;
+    const SUBPAGE: u32 = 0x0;
+    const EBX_BIT: u32 = 1 << 25;
+    let ebx_out: u32;
+
+    unsafe {
+        asm!(r"
+              mov $1, %eax;
+              mov $2, %ecx;
+              cpuid;"
+            : "={ebx}" (ebx_out)
+            : "i" (LEAF), "i" (SUBPAGE)
+            : "eax", "ecx", "edx"
+            : "volatile");
+    }
+
+    if ebx_out & (EBX_BIT) != 0 {
```

Comment:
Simplify to:

```rust
ebx_out & EBX_BIT != 0
```

Reply:
Not sure why that's like that. I will update the source of the paste too.